starlette/datastructures.py

```diff
@@ -415,15 +415,22 @@ class UploadFile:
     """
 
     spool_max_size = 1024 * 1024
+    headers: "Headers"
 
     def __init__(
-        self, filename: str, file: typing.IO = None, content_type: str = ""
+        self,
+        filename: str,
+        file: typing.IO = None,
+        content_type: str = "",
+        *,
+        raw_headers: typing.Optional[typing.List[typing.Tuple[bytes, bytes]]] = None,
```

Comment:
I think it'd make more sense to me if the API here was `headers: Headers`.

The `formparsers.py` module could be responsible for instantiating the headers from the raw items, right?

Reply:
Yep great call, not sure why I wasn't doing that from the get go